### PR TITLE
Clean up unit and integration test for kubeflow

### PR DESCRIPTION
**Pull request by alculquicondor**
### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

- Add a unit test for applying node selectors on admission
- Clean up integration tests

#### Which issue(s) this PR fixes:

As part of investigation for #1407

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/controller/jobs/kubeflow/jobs/mxjob/mxjob_controller_test.go [hunk 1]
```diff
@@ -311,10 +312,12 @@ func TestReconciler(t *testing.T) {
 	cases := map[string]struct {
 		reconcilerOptions []jobframework.Option
 		job               *kftraining.MXJob
+		flavors           []kueue.ResourceFlavor
 		workloads         []kueue.Workload
 		wantJob           *kftraining.MXJob
 		wantWorkloads     []kueue.Workload
 		wantErr           error
+		focus             bool
```
Comment (tenzen-y) [inline pkg/controller/jobs/kubeflow/jobs/mxjob/mxjob_controller_test.go:320]:
```suggestion
```

It probably isn't needed?

File: pkg/util/testingjobs/mxjob/wrappers.go [hunk 2]
```diff
@@ -175,15 +175,17 @@ func (j *MXJobWrapper) UID(uid string) *MXJobWrapper {
...
 // NodeSelector updates the nodeSelector of job.
 func (j *MXJobWrapper) NodeSelector(k, v string) *MXJobWrapper {
-	if j.Spec.MXReplicaSpecs[kftraining.MXJobReplicaTypeServer].Template.Spec.NodeSelector == nil {
-		j.Spec.MXReplicaSpecs[kftraining.MXJobReplicaTypeServer].Template.Spec.NodeSelector = make(map[string]string)
-	}
-	if j.Spec.MXReplicaSpecs[kftraining.MXJobReplicaTypeWorker].Template.Spec.NodeSelector == nil {
-		j.Spec.MXReplicaSpecs[kftraining.MXJobReplicaTypeWorker].Template.Spec.NodeSelector = make(map[string]string)
+	return j.
+		RoleNodeSelector(kftraining.MXJobReplicaTypeServer, k, v).
+		RoleNodeSelector(kftraining.MXJobReplicaTypeWorker, k, v)
```
Comment (tenzen-y) [inline pkg/util/testingjobs/mxjob/wrappers.go:180]:
```suggestion
	return j.RoleNodeSelector(kftraining.MXJobReplicaTypeServer, k, v).
		RoleNodeSelector(kftraining.MXJobReplicaTypeWorker, k, v)
```

**Review by tenzen-y — COMMENTED** [review]
LGTM

I left comments for some nits.

File: pkg/util/testingjobs/mxjob/wrappers.go (hunk 2 above)
Comment (alculquicondor) [reply pkg/util/testingjobs/mxjob/wrappers.go]: done

File: pkg/controller/jobs/kubeflow/jobs/mxjob/mxjob_controller_test.go (hunk 1 above)
Comment (alculquicondor) [reply pkg/controller/jobs/kubeflow/jobs/mxjob/mxjob_controller_test.go]: right, removed

**Pull request merged**